2016/proposals/filip-ter.md

```diff
@@ -0,0 +1,197 @@
+##Providing sparse data structures as an implementation detail of the core and enhancing their functionality
+
+---
+
+###Table of Contents
+
+
+ 1. Abstract
+ 2. Project Goals
+ 3. Technical Details
+ 4. Milestones / Schedule of Deliverables
+ 5. Future Works
+ 6. Open Source Development Experience
+ 7. Academic Experience
+ 8. Why this project
+ 9. About Me
+ 10. Appendix
+ 
+---
+
+###Abstract
+
+Currently sparse data structures provided by pandas exist in the separate `sparse` module. This package has few users and is easy to overlook when developing new features due to its separation from the `core` module. I propose to change sparse into an implementation detail of the core data structures so that a user does not need to import different packages or classes if they want to use sparse functionality. 
+
+As it stands now, sparse data structures are meant to be compatible with the general data structures they represent. Making sparse integrated with the `core` would make it easier to deal with compatibility and require less changes from users who want to change their code to use sparse.
+
+The `sparse` module also has many issues that have yet to be resolved, as can be seen [here](https://github.com/pydata/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ASparse). The errors make sparse more difficult to use and require workarounds. On top of this there are enhancements to sparse that have been requested on Github. Once the refactoring is finished, I plan to fix some of the more relevant bugs and implement the enhancements, in order to provide a wider range of features to sparse data structures. 
```

Comment:
I would focus here, just making sparse integration better. I think that you will learn from the fixing of bugs enough to have a better picture of sparse, where it fits and what should be done with it.